### ISSUE TITLE
A duplicate error is reported for an ambiguous user-defined conversion

### ISSUE THREAD

**Issue by AlekseyTs** [dotnet/roslyn#81984]
```
public struct S1
{
    public S1(S2 x) => throw null;
    public S1(string x) => throw null;

    public static implicit operator S1(S2 x) => throw null;
}

public struct S2
{
    public static implicit operator S1(S2 x) => throw null;
}

class Program
{
    static S1 Test2(S2 x)
    {
        return x;
    }   
}
```

Observed:
```
// (20,16): error CS0457: Ambiguous user defined conversions 'S2.implicit operator S1(S2)' and 'S1.implicit operator S1(S2)' when converting from 'S2' to 'S1'
//         return x;
Diagnostic(ErrorCode.ERR_AmbigUDConv, "x").WithArguments("S2.implicit operator S1(S2)", "S1.implicit operator S1(S2)", "S2", "S1").WithLocation(20, 16),
// (20,16): error CS0457: Ambiguous user defined conversions 'S2.implicit operator S1(S2)' and 'S1.implicit operator S1(S2)' when converting from 'S2' to 'S1'
//         return x;
Diagnostic(ErrorCode.ERR_AmbigUDConv, "x").WithArguments("S2.implicit operator S1(S2)", "S1.implicit operator S1(S2)", "S2", "S1").WithLocation(20, 16)
```
